lib/charms/kubernetes_charm_libraries/v0/volumes.py

```diff
@@ -0,0 +1,415 @@
+            pod = self.client.get(Pod, name=pod_name, namespace=self.namespace)
+        except ApiError as e:
+            if e.status.reason == "Unauthorized":
+                logger.debug("kube-apiserver not ready yet")
+            else:
+                raise KubernetesRequestedVolumesError(f"Pod {pod_name} not found")
+            return False
+        return self._pod_contains_requested_volumes(
+            pod=pod,  # type: ignore[arg-type]
+            requested_volumes=requested_volumes,
+            container_name=container_name,
+        )
+
+    def statefulset_is_patched(
+        self,
+        statefulset_name: str,
+        requested_volumes: Iterable[RequestedVolume],
+    ) -> bool:
+        """Returns whether the statefulset has the expected requested volumes.
+
+        Args:
+            statefulset_name: Statefulset name.
+            requested_volumes: Iterable of requested volumes
+
+        Returns:
+            bool: Whether the statefulset has the expected requested volumes.
+        """
+        try:
+            statefulset = self.client.get(
+                res=StatefulSet, name=statefulset_name, namespace=self.namespace
+            )
+        except ApiError as e:
+            if e.status.reason == "Unauthorized":
+                logger.debug("kube-apiserver not ready yet")
+            else:
+                raise KubernetesRequestedVolumesError(
+                    f"Could not get statefulset {statefulset_name}"
+                )
+            return False
+        return self._statefulset_volumes_are_patched(
+            statefulset_spec=statefulset.spec,  # type: ignore[attr-defined]
+            requested_volumes=requested_volumes,
+        )
+
+    @staticmethod
+    def _statefulset_volumes_are_patched(
+        statefulset_spec: StatefulSetSpec,
+        requested_volumes: Iterable[RequestedVolume],
+    ) -> bool:
+        """Returns whether a StatefulSet is patched with requested volumes.
+
+        Args:
+            statefulset_spec: StatefulSet spec
+            requested_volumes: Iterable of requested volumes
+
+        Returns:
+            bool
+        """
+        if not statefulset_spec.template.spec.volumes:
+            return False
+        return all(
+            [
+                requested_volume.volume in statefulset_spec.template.spec.volumes
+                for requested_volume in requested_volumes
+            ]
+        )
+
+    def _pod_contains_requested_volumes(
+        self,
+        container_name: str,
+        requested_volumes: Iterable[RequestedVolume],
+        pod: Union[PodTemplateSpec, Pod],
+    ) -> bool:
+        """Returns whether a pod is patched with requested volumes.
+
+        Args:
+            container_name: Container name
+            requested_volumes: Iterable of requested volumes
+            pod: Kubernetes pod object.
+
+        Returns:
+            bool
+        """
+        return self._pod_volumemounts_contain_requested_volumes(
```

Comment:
There is no need to forward the parameters to another method call. 
_pod_contains_requested_volumes and _pod_volumemounts_contain_requested_volumes could be just one method.

Reply:
True, however I just realized that if we are checking a HugePages volume we need to check both that volumeMount exist and that ResourceRequirements are correctly set; in that case I would add another method for that (`_pod_resources_are_set`) if you agree